src/main/java/com/google/devtools/build/lib/rules/java/JavaHeaderCompileAction.java

```diff
@@ -499,7 +522,8 @@ public void build(JavaToolchainProvider javaToolchain) throws RuleErrorException
                 // If classPathMode == BAZEL, also make sure to inject the dependencies to be
                 // available to downstream actions. Else just do enough work to locally create the
                 // full .jdeps from the .stripped .jdeps produced on the executor.
-                /* insertDependencies= */ classpathMode == JavaClasspathMode.BAZEL));
+                /* insertDependencies= */ classpathMode == JavaClasspathMode.BAZEL,
+                additionalArtifactsForPathMapping));
```

Comment:
As I understand, this adds a new reference to the transitive class path to `JavaheaderCompileAction`. Does that have any memory consequences in that the reference could live longer? I realize there's already a reference passed to `JavaCompileAction` (not sure it's the same reference, since `JavaHeaderCompileActionBuilder` only constructs a `JavaCompileAction` under certain circumstances). Maybe that reference eliminates this concern?

Reply:
I verified with JOL that the instance size of `JavaHeaderCompileAction` is still 72 after this change (it had 7 bytes of padding previously), so the field itself shouldn't result in higher memory usage. @justinhorvitz Is "HotSpot >= 15, 64-bit, COOPS, CCPS" the correct JOL setting to analyze retained size for Bazel?

The nested set it creates an additional reference to is either empty or comes from `JavaTargetAttributes#getCompileTimeClassPath()`, which I think is always retained by the corresponding full compilation action here: https://github.com/bazelbuild/bazel/blob/379ee5fd888c8f3a83ed3751bee700287fa5d7de/src/main/java/com/google/devtools/build/lib/rules/java/JavaCompilationHelper.java#L331-L346.